### PR TITLE
Add application proposal for Karmada

### PR DESCRIPTION
**Pull request by Poor12**
Signed-off-by: Poor12 <shentiecheng@huawei.com>

**What type of PR is this?**
/kind design
/kind documentation

**What this PR does / why we need it**:
Add application proposal for Karmada to enable users to manage multi-cluster applications in Karmada.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
cc @RainbowMango  @kevin-wangzefeng 

**Does this PR introduce a user-facing change?**:

### CODE REVIEW

**Comment by Fish-pro** [comment]
@Poor12 I would like to ask you why the multi-cluster application does not include the PropagationPolicy and OverridePolicy, how to consider this, are there any more learning documents about application?

**Comment by Poor12** [comment]
> @Poor12I would like to ask you why the multi-cluster application does not include the PropagationPolicy and OverridePolicy, how is this considered? Is there more documentation about the application that I can learn?

Application is more like workloads with more complex organizational forms. I'm more inclined to follow karmada's way of dealing with kubernetes objects. This feature is still on design. We can discuss on the community meeting soon.

File: docs/proposals/application/README.md [hunk 1]
```diff
@@ -0,0 +1,571 @@
...
+	ComponentGroupKinds []metav1.GroupKind `json:"componentKinds,omitempty"`
+
+	// Descriptor regroups information and metadata about an application.
+	Descriptor Descriptor `json:"descriptor,omitempty"`
+
+	// Selector is a label query over kinds that created by the application. It must match the component objects' labels.
+	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors
+	Selector *metav1.LabelSelector `json:"selector,omitempty"`
+
+	// Info contains human readable key,value pairs for the Application.
```
Comment (Tingtal) [inline docs/proposals/application/README.md:346]:
```suggestion
	// Info contains human readable key-value pairs for the Application.
```

File: docs/proposals/application/README.md [hunk 2]
```diff
@@ -0,0 +1,571 @@
...
+	Maintainers []ContactData `json:"maintainers,omitempty"`
+
+	// Owners is an optional list of the owners of the installed application. The owners of the application should be
+	// contacted in the event of a planned or unplanned disruption affecting the application.
+	Owners []ContactData `json:"owners,omitempty"`
+
+	// Keywords is an optional list of key words associated with the application (e.g. MySQL, RDBMS, database).
+	Keywords []string `json:"keywords,omitempty"`
+
+	// Links are a list of descriptive URLs intended to be used to surface additional documentation, dashboards, etc.
```
Comment (Tingtal) [inline docs/proposals/application/README.md:378]:
```suggestion
	// Links is a list of descriptive URLs intended for surfacing additional documentation, dashboards, etc.
```

File: docs/proposals/application/README.md [hunk 3]
```diff
@@ -0,0 +1,571 @@
...
+// ImageSpec contains information about an image used as an icon.
+type ImageSpec struct {
+	// The source for image represented as either an absolute URL to the image or a Data URL containing
+	// the image. Data URLs are defined in RFC 2397.
+	Source string `json:"src"`
+
+	// (optional) The size of the image in pixels (e.g., 25x25).
+	Size string `json:"size,omitempty"`
+
+	// (optional) The mine type of the image (e.g., "image/png").
```
Comment (Tingtal) [inline docs/proposals/application/README.md:395]: Should be "mime" type?

File: docs/proposals/application/README.md [hunk 4]
```diff
@@ -0,0 +1,571 @@
...
+// Link contains information about an URL to surface documentation, dashboards, etc.
+type Link struct {
+	// Description is human readable content explaining the purpose of the link.
+	Description string `json:"description,omitempty"`
+
+	// Url typically points at a website address.
+	URL string `json:"url,omitempty"`
+}
+
+// InfoItem is a human readable key,value pair containing important information about how to access the Application.
```
Comment (Tingtal) [inline docs/proposals/application/README.md:420]:
```suggestion
// InfoItem is a human readable key-value pair containing important information about how to access the Application.
```

File: docs/proposals/application/README.md [hunk 5]
```diff
@@ -0,0 +1,571 @@
...
+        resource: ""
+        version: v1
+      status:
+        applied: true
+        healthy: true
+```
+
+### Test Plan
+
+- Unit Test covering.
```
Comment (Tingtal) [inline docs/proposals/application/README.md:570]: Not sure what "covering" means here.
In my opinion, "Unit tests" is enough. Pls check.

File: docs/proposals/application/README.md [hunk 6]
```diff
@@ -0,0 +1,571 @@
...
+- E2E covering.
```
Comment (Tingtal) [inline docs/proposals/application/README.md:571]: E2E coverage?

**Review by Tingtal — CHANGES_REQUESTED** [review]
Small touches on the language. Pls check.

**Comment by nuclearwu** [comment]
> > @Poor12I would like to ask you why the multi-cluster application does not include the PropagationPolicy and OverridePolicy, how is this considered? Is there more documentation about the application that I can learn?
> 
> Application is more like workloads with more complex organizational forms. I'm more inclined to follow karmada's way of dealing with kubernetes objects. This feature is still on design. We can discuss on the community meeting soon.

At present, application distribution mainly includes two scenarios for us, one is to deploy differentiated applications in different clusters, and the other is to distribute Kubernetes resources in applications containing multiple workloads to different clusters. So I wonder if karmada will support both? Also, can we get involved together?

**Comment by Poor12** [comment]
Hi, @nuclearwu, Application is a budding idea. We actually want to listen more users's voice. If you have any demands or ideas, welcome to talk about it in our slack channel or wechat group.

**Pull request closed without merging**